src/journal/journal-file.c

```diff
@@ -464,6 +497,9 @@ static int journal_file_refresh_header(JournalFile *f) {
         /* Sync the online state to disk */
         (void) fsync(f->fd);
 
+        /* We likely just created a new file, also sync the directory this file is located in. */
+        (void) fsync_directory_of_file(f->fd);
```

Comment:
That's unrelated to this change. Basically, we sync the file, then mark the file as clean in its header and sync again. The marking as clean you can't see between the two syncs of course.